kyo-scheduler/jvm/src/main/scala/kyo/scheduler/Scheduler.scala

```diff
@@ -70,7 +70,7 @@ final class Scheduler(
         var worker: Worker = null
         if (submitter eq null) {
             worker = Worker.current()
-            if ((worker ne null) && !worker.checkAvailability(nowMs))
+            if ((worker ne null) && ((worker eq submitter) || !worker.checkAvailability(nowMs)))
```

Comment:
This check was missing here, making the worker try to enqueue to itself while draining.